src/lib.rs

```diff
@@ -336,4 +372,37 @@ mod tests {
             assert_eq!(f2.decrypt(&f1.encrypt(&val)), Ok(val.clone()));
         }
     }
+
+    #[test]
+    fn test_multi_encrypt() {
+        let f1 = Fernet::new(&Fernet::generate_key()).unwrap();
+        let f2 = Fernet::new(&Fernet::generate_key()).unwrap();
+        let f = MultiFernet::new(vec![f1, f2]);
+        assert_eq!(f1.decrypt(&f.encrypt(b"abc")).unwrap(), b"abc".to_vec());
```

Comment:
Can you add an `assert_eq!(f2.decrypt(&f.decrypt(b"abc")), Err(DecryptionError));`?

last comment, I swear :-)

Reply:
No, but I can add ```assert_eq!(f2.decrypt(&f.encrypt(b"abc")), Err(DecryptionError));```. ;)

Reply:
:D